source/server/options_impl.cc

```diff
@@ -64,6 +72,13 @@ OptionsImpl::OptionsImpl(int argc, char** argv, const std::string& hot_restart_v
                                     "One of 'serve' (default; validate configs and then serve "
                                     "traffic normally) or 'validate' (validate configs and exit).",
                                     false, "serve", "string", cmd);
+  TCLAP::ValueArg<uint64_t> max_stats("", "max-stats",
+                                      "Maximum number of stats guages and counters "
+                                      "that can be allocated in shared memory.",
+                                      false, ENVOY_DEFAULT_MAX_STATS, "uint64_t", cmd);
+  TCLAP::ValueArg<uint64_t> max_stat_name_len("", "max-stat-name-len",
+                                              "Maximum name length for a stat", false,
```

Comment:
I am missing something here. We are not making the cluster Name configurable (its 60). So why bother making this name configurable since the current 128byte length seems to handle all the stats? Unless you happen to have custom filters that generate stats exceeding this length.

Reply:
In a follow up change we will make all the user supplied name max lengths dynamic based on this parameter.